test_util/cluster_api.py

```diff
@@ -327,31 +326,28 @@ def cluster_api_request(self, method, path, host_url=None, node=None, port=None,
         if node is not None:
             assert host_url is None, 'Cannot set both node and host_url'
             host_url = self.get_node_url(node, port=port)
-        return self.api_request(method, path, host_url=host_url, **kwargs)
-
-    get = functools.partialmethod(cluster_api_request, 'get')
-    post = functools.partialmethod(cluster_api_request, 'post')
-    put = functools.partialmethod(cluster_api_request, 'put')
-    delete = functools.partialmethod(cluster_api_request, 'delete')
-    options = functools.partialmethod(cluster_api_request, 'options')
-    head = functools.partialmethod(cluster_api_request, 'head')
-    patch = functools.partialmethod(cluster_api_request, 'patch')
-    delete = functools.partialmethod(cluster_api_request, 'delete')
+        return orig_wrapper(request_fn, path, host_url=host_url, **kwargs)
 
     def get_client(self, path, default_headers=None):
-        return test_util.helpers.ApiClient(
+        new_client = test_util.helpers.ApiClient(
             default_host_url=self.dcos_url,
             api_base=path,
             ca_cert_path=self.ca_cert_path,
             default_headers=self.default_headers if default_headers is None else default_headers)
+        # give spawned clients the node= reqest option
+        new_client.extra_wrapper = self.extra_wrapper
+        return new_client
 
     @property
     def marathon(self):
-        return test_util.marathon.Marathon(
+        marathon_client = test_util.marathon.Marathon(
```

Comment:
Also note for that, might be possible to just do one ApiClient, making since making the `get_node_url` a parameter gets rid of the circular reference.